extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/WorkerChatHandler.java

```diff
@@ -193,10 +194,17 @@ public Response httpFetchKeyStatistics(
     ChatHandlers.authorizationCheck(req, Action.READ, task.getDataSource(), toolbox.getAuthorizerMapper());
     ClusterByStatisticsSnapshot clusterByStatisticsSnapshot;
     StageId stageId = new StageId(queryId, stageNumber);
-    clusterByStatisticsSnapshot = worker.fetchStatisticsSnapshot(stageId);
-    return Response.status(Response.Status.ACCEPTED)
-                   .entity(clusterByStatisticsSnapshot)
-                   .build();
+    try {
+      clusterByStatisticsSnapshot = worker.fetchStatisticsSnapshot(stageId);
+      return Response.status(Response.Status.ACCEPTED)
+                     .entity(clusterByStatisticsSnapshot)
+                     .build();
+    }
+    catch (ISE e) {
+      log.error(e, "Invalid request for key statistics");
```

Comment:
Should we return the error message to the client ?

Reply:
We can do that, I wasn't sure if this needed to be returned to the caller or just logged. I would also assume that we should call ISE.sanitize() before we do so, correct?

Reply:
Yup.